templates/Makefile.erb

```diff
@@ -79,8 +79,11 @@ cleanup-components: <%= @components.map {|comp| "#{comp.name}-cleanup" }.join("
 	touch <%= comp.name %>-patch
 
 <%= comp.name %>-configure: <%= comp.name %>-patch <%= list_component_dependencies(comp).join(" ") %>
+	<%- if comp.get_build_dir -%>
+	[ -d <%= comp.get_build_dir %> ] || mkdir <%= comp.get_build_dir %>
```

Comment:
Ah, here we go:

```
mckern@flexo vanagon $ type mkdir
mkdir is aliased to `mkdir -p'
mckern@flexo vanagon $ \mkdir bin
mkdir: cannot create directory ‘bin’: File exists
```

So, `mkdir` fails if a directory exists but `mkdir -p` does not. Today we learned something.